From 73aa43765175b4577b01682bc756f273ea19d82f Mon Sep 17 00:00:00 2001
From: Dennis Gilmore <dennis@ausil.us>
Date: Fri, 31 Aug 2012 16:38:20 -0500
Subject: [PATCH] ensure that we dont already have a /dev/loopX in the chroot
 before making one

---
 builder/kojid | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/builder/kojid b/builder/kojid
index 58720b4..24fbecc 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -1771,7 +1771,7 @@ class ImageTask(BaseTaskHandler):
         broot.init()
 
         # Create the loopback devices we need
-        cmd = 'for i in $(seq 0 7); do mknod /dev/loop$i b 7 $i; done'
+        cmd = 'for i in $(seq 0 7); do [ ! -e /dev/loop$i ] && (mknod /dev/loop$i b 7 $i); done'
         rv = broot.mock(['--chroot', cmd])
         if rv:
             broot.expire()
-- 
1.7.11.4

