--- koji-1.4.0/cli/koji.conf.orig	2010-07-08 21:04:26.000000000 -0500
+++ koji-1.4.0/cli/koji.conf	2010-07-09 10:21:24.000000000 -0500
@@ -3,24 +3,24 @@
 ;configuration for koji cli tool
 
 ;url of XMLRPC server
-;server = http://hub.example.com/kojihub
+server = http://koji.fedoraproject.org/kojihub
 
 ;url of web interface
-;weburl = http://www.example.com/koji
+weburl = http://koji.fedoraproject.org/koji
 
 ;url of package download site
-;pkgurl = http://www.example.com/packages
+pkgurl = http://koji.fedoraproject.org/packages
 
 ;path to the koji top directory
-;topdir = /mnt/koji
+topdir = /mnt/koji
 
 ;configuration for SSL authentication
 
 ;client certificate
-;cert = ~/.koji/client.crt
+cert = ~/.fedora.cert
 
 ;certificate of the CA that issued the client certificate
-;ca = ~/.koji/clientca.crt
+ca = ~/.fedora-server-ca.cert
 
 ;certificate of the CA that issued the HTTP server certificate
-;serverca = ~/.koji/serverca.crt
+serverca = ~/.fedora-server-ca.cert
