diff -uNr koji-1.7.1-orig/builder/kojid koji-1.7.1/builder/kojid
--- koji-1.7.1-orig/builder/kojid	2012-11-19 22:36:03.000000000 -0600
+++ koji-1.7.1/builder/kojid	2013-01-20 10:08:08.408515512 -0600
@@ -2698,7 +2698,7 @@
 
     def create_local_repo(self, rinfo, arch, pkglist, groupdata, oldrepo):
         koji.ensuredir(self.outdir)
-        cmd = ['/usr/bin/createrepo', '-vd', '-o', self.outdir]
+        cmd = ['/usr/bin/createrepo', '-vd', '-o', self.outdir, '-u', self.options.topurl]
         if pkglist is not None:
             cmd.extend(['-i', pkglist])
         if os.path.isfile(groupdata):
@@ -2725,7 +2725,7 @@
         if pkglist is None:
             cmd.append(self.outdir)
         else:
-            cmd.append(self.repodir)
+            cmd.append(self.options.topdir)
 
         logfile = '%s/createrepo.log' % self.workdir
         status = log_output(self.session, cmd[0], cmd, logfile, self.getUploadDir(), logerror=True)
diff -uNr koji-1.7.1-orig/hub/kojihub.py koji-1.7.1/hub/kojihub.py
--- koji-1.7.1-orig/hub/kojihub.py	2012-11-19 22:36:03.000000000 -0600
+++ koji-1.7.1/hub/kojihub.py	2013-01-20 10:08:11.076535467 -0600
@@ -2188,7 +2188,7 @@
     packages = {}
     for repoarch in repo_arches:
         packages.setdefault(repoarch, [])
-    relpathinfo = koji.PathInfo(topdir='toplink')
+    relpathinfo = koji.PathInfo(topdir='')
     for rpminfo in rpms:
         if not with_debuginfo and koji.is_debuginfo(rpminfo['name']):
             continue
@@ -2221,10 +2221,6 @@
             # src and noarch special-cased -- see below
         archdir = os.path.join(repodir, arch)
         koji.ensuredir(archdir)
-        # Make a symlink to our topdir
-        top_relpath = koji.util.relpath(koji.pathinfo.topdir, archdir)
-        top_link = os.path.join(archdir, 'toplink')
-        os.symlink(top_relpath, top_link)
         pkglist = file(os.path.join(repodir, arch, 'pkglist'), 'w')
         logger.info("Creating package list for %s" % arch)
         for rpminfo in packages[arch]:
